qiskit/aqua/algorithms/phase_estimators/hamiltonian_pe_result.py

```diff
@@ -0,0 +1,96 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Result from running HamiltonianPE"""
+
+
+from typing import Dict, List, cast
+from qiskit.result import Result
+from .phase_estimator_result import PhaseEstimatorResult
+from .phase_estimation_scale import PhaseEstimationScale
+
+
+class HamiltonianPEResult(PhaseEstimatorResult):
+    """Store and manipulate results from running `HamiltonianPE`.
+
+    This API of this class is nearly the same as `PhaseEstimatorResult`, differing only in
+    the presence of an additional keyword argument in the methods. If `scaled`
+    is `False`, then the phases are not translated and scaled to recover the
+    eigenvalues of the Hamiltonian. Instead `phi` in `[0, 1)` is returned,
+    as is the case when then unitary is not derived from a Hamiltonian.
+    """
+    def __init__(self, num_evaluation_qubits: int, circuit_result: Result,
+                 phase_estimation_scale: PhaseEstimationScale,
+                 phase_array: List = None, phase_dict: Dict = None) -> None:
```

Comment:
Can you make the type hints more precise and add the `Optional` type? I.e.
`Optional[List[float]] = None` instead of `List = None`.